.github/workflows/spread-tests.yaml

```diff
@@ -0,0 +1,274 @@
+          fi
+
+          # This coud be the case when either there are not systems for a group or 
+          # the list of tests to run is empty
+          if [ -z "$RUN_TESTS" ]; then
+            echo "No tests to run, skiping..."
+            exit 0
+          fi
+
+          # Run spread tests
+          # "pipefail" ensures that a non-zero status from the spread is
+          # propagated; and we use a subshell as this option could trigger
+          # undesired changes elsewhere
+          echo "Running command: $SPREAD $RUN_TESTS"
+          (set -o pipefail; $SPREAD -no-debug-output -logs spread-logs $RUN_TESTS | PYTHONDONTWRITEBYTECODE=1 ./tests/lib/external/snapd-testing-tools/utils/log-filter $FILTER_PARAMS | tee spread.log)
+
+    - name: Upload spread logs
+      if: always()
+      uses: actions/upload-artifact@v4
+      with:
+        name: spread-logs-${{ inputs.systems }}
+        path: "spread-logs/*.log"
+        if-no-files-found: ignore
+
+    - name: Discard spread workers
+      if: always()
+      run: |
+        shopt -s nullglob;
+        for r in .spread-reuse.*.yaml; do
+          spread -discard -reuse-pid="$(echo "$r" | grep -o -E '[0-9]+')";
+        done
+
+    - name: Report spread errors
+      if: always()
+      run: |
+        if [ -e spread.log ]; then
+            echo "Running spread log analyzer"
+            ACTIONS_URL="${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}"
+            PYTHONDONTWRITEBYTECODE=1 ./tests/lib/external/snapd-testing-tools/utils/log-parser spread.log --output spread-results.json --cut 1 >/dev/null
+            while IFS= read -r line; do
+                if [ ! -z "$line" ]; then
+                    echo "Adding failed test line to filtered log"
+                    echo "GRAFANA FAILED: $line $ACTIONS_URL" | tee -a "$FILTERED_LOG_FILE"
+                fi
+            done <<< $(jq -r '.[] | select( .type == "info" ) | select( .info_type == "Error" ) | "\(.verb) \(.task)"' spread-results.json)
+        else
+            echo "No spread log found, skipping errors reporting"
+        fi
+
+    - name: Analyze spread test results
+      if: always()
+      run: |
+          if [ -f spread.log ]; then
+              echo "Running spread log parser"
+              PYTHONDONTWRITEBYTECODE=1 ./tests/lib/external/snapd-testing-tools/utils/log-parser spread.log --output spread-results.json
+
+              # Add openstack backend definition to spread.yaml
+              if [ "${{ inputs.backend }}" = openstack ]; then
+                  ./tests/lib/spread/add-backend tests/lib/spread/backend.openstack.yaml spread.yaml
+              fi
+
+              echo "Running spread log analyzer"
+              PYTHONDONTWRITEBYTECODE=1 ./tests/lib/external/snapd-testing-tools/utils/log-analyzer list-reexecute-tasks "$RUN_TESTS" spread-results.json > "$FAILED_TESTS_FILE"
```

Comment:
I suppose, `PYTHONDONTWRITEBYTECODE=1` could be set for the whole workflow. We can try this in a followup PR.
Also, I have hard time trying to remember why we even set it. Might have been that some of the *.pyc files are picked up by spread or static-checks?

Reply:
I'll look into this in the follow-up. That's a good question.